From 403111b2f49f6d2a7ba1ad55030c3690b6187b4f Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Mon, 25 Dec 2017 20:57:28 +0800
Subject: [PATCH] FROMLIST: [media] dt/bindings: Add bindings for OV5695

Add device tree binding documentation for the OV5695 sensor.

Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>

BUG=b:36227021
TEST=Camera works on Scarlet with the whole series
(am from https://patchwork.linuxtv.org/patch/46249/)

Change-Id: Id4c9b1c6a576f82a1c6a0119aa2017c46a552b46
Signed-off-by: Jacob Chen <jacob2.chen@rock-chips.com>
---
 .../devicetree/bindings/media/i2c/ov5695.txt  | 38 +++++++++++++++++++
 1 file changed, 38 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/i2c/ov5695.txt

diff --git a/Documentation/devicetree/bindings/media/i2c/ov5695.txt b/Documentation/devicetree/bindings/media/i2c/ov5695.txt
new file mode 100644
index 000000000000..8d87dbcdef70
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/ov5695.txt
@@ -0,0 +1,38 @@
+* Omnivision OV5695 MIPI CSI-2 sensor
+
+Required Properties:
+- compatible: should be "ovti,ov5695"
+- clocks: reference to the 24M xvclk input clock.
+- clock-names: should be "xvclk".
+- dovdd-supply: Digital I/O voltage supply, 1.8 volts
+- avdd-supply: Analog voltage supply, 2.8 volts
+- dvdd-supply: Digital core voltage supply, 1.2 volts
+- reset-gpios: Low active reset gpio
+
+The device node must contain one 'port' child node for its digital output
+video port, in accordance with the video interface bindings defined in
+Documentation/devicetree/bindings/media/video-interfaces.txt.
+
+Example:
+&i2c1: camera-sensor@36 {
+	compatible = "ovti,ov5695";
+	reg = <0x36>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&clk_24m_cam>;
+
+	clocks = <&cru SCLK_TESTCLKOUT1>;
+	clock-names = "xvclk";
+
+	avdd-supply = <&pp2800_cam>;
+	dvdd-supply = <&pp1250_cam>;
+	dovdd-supply = <&pp1800>;
+
+	reset-gpios = <&gpio2 5 GPIO_ACTIVE_LOW>;
+
+	port {
+		wcam_out: endpoint {
+			remote-endpoint = <&mipi_in_wcam>;
+			data-lanes = <1 2>;
+		};
+	};
+};
-- 
2.35.3

